src/main/scala/org/apache/spark/sql/kinesis/KinesisReader.scala

```diff
@@ -168,6 +169,22 @@ private[kinesis] case class KinesisReader(
           getAmazonClient.describeStream(describeStreamRequest)
       }
     }
+
+    val shards = new ArrayList[Shard]()
+    var exclusiveStartShardId : String = null
+
+    do {
+        describeStreamRequest.setExclusiveStartShardId( exclusiveStartShardId )
+        val describeStreamResult = getAmazonClient.describeStream( describeStreamRequest )
+        shards.addAll( describeStreamResult.getStreamDescription().getShards() )
+        if (describeStreamResult.getStreamDescription().getHasMoreShards() && shards.size() > 0) {
+          exclusiveStartShardId = shards.get(shards.size() - 1).getShardId();
+        } else {
+          exclusiveStartShardId = null
+       }
+    } while ( exclusiveStartShardId != null )
+
+    /*
     val streamDescription = describeStreamResult.getStreamDescription()
```

Comment:
Shall we delete these lines instead of commenting them

Reply:
Also we can delete `private val maxSupportedShardsPerStream = 100` in line number 70

Reply:
Yes, we can delete the commented lines. I just left them in there for reference. And yes, we can delete `private val maxSupportedShardsPerStream = 100` on line 70.

Reply:
Sure. Can you delete the lines so that I can merge your PR